### PR TITLE
Uniform access to code analysis via ExecutionState

### PR DESCRIPTION
**Pull request by chfast**
Add pointer to code analysis to ExecutionState to unify signature of instruction implementations.

### CODE REVIEW

**Comment by chfast** [comment]
```
advanced/execute/main/blake2b_huff/empty_mean                         -0.0126         -0.0126            13            13            13            13                   
baseline/execute/main/blake2b_huff/empty_mean                         +0.0215         +0.0215            16            16            16            16                   
advanced/execute/main/blake2b_huff/2805nulls_mean                     -0.0132         -0.0132           284           281           284           281                   
baseline/execute/main/blake2b_huff/2805nulls_mean                     +0.0298         +0.0298           343           353           343           353                   
advanced/execute/main/blake2b_huff/5610nulls_mean                     -0.0203         -0.0203           558           547           558           547                   
baseline/execute/main/blake2b_huff/5610nulls_mean                     +0.0254         +0.0253           670           687           670           687                   
advanced/execute/main/blake2b_huff/8415nulls_mean                     -0.0170         -0.0170           816           802           816           802                   
baseline/execute/main/blake2b_huff/8415nulls_mean                     +0.0319         +0.0319           983          1014           983          1014                   
advanced/execute/main/blake2b_huff/65536nulls_mean                    -0.0084         -0.0084          6299          6246          6299          6246                   
baseline/execute/main/blake2b_huff/65536nulls_mean                    +0.0339         +0.0339          7644          7903          7644          7903                   
advanced/execute/main/blake2b_shifts/2805nulls_mean                   -0.0089         -0.0089          3720          3687          3720          3687                   
baseline/execute/main/blake2b_shifts/2805nulls_mean                   +0.0124         +0.0124          3254          3294          3254          3294                   
advanced/execute/main/blake2b_shifts/5610nulls_mean                   -0.0057         -0.0057          7404          7362          7404          7362                   
baseline/execute/main/blake2b_shifts/5610nulls_mean                   +0.0070         +0.0070          6520          6566          6520          6566                   
advanced/execute/main/blake2b_shifts/8415nulls_mean                   -0.0089         -0.0089         11119         11020         11119         11021                   
baseline/execute/main/blake2b_shifts/8415nulls_mean                   -0.0026         -0.0026          9842          9817          9842          9817                   
advanced/execute/main/blake2b_shifts/65536nulls_mean                  -0.0068         -0.0068         86232         85647         86232         85646                   
baseline/execute/main/blake2b_shifts/65536nulls_mean                  -0.0040         -0.0040         76395         76091         76395         76091                   
advanced/execute/main/sha1_divs/empty_mean                            +0.0063         +0.0063            56            56            56            56                   
baseline/execute/main/sha1_divs/empty_mean                            +0.0081         +0.0081            59            59            59            59                   
advanced/execute/main/sha1_divs/1351_mean                             +0.0076         +0.0076          1155          1164          1155          1164                   
baseline/execute/main/sha1_divs/1351_mean                             +0.0067         +0.0067          1231          1240          1231          1240
advanced/execute/main/sha1_divs/2737_mean                             +0.0068         +0.0068          2256          2271          2256          2271
baseline/execute/main/sha1_divs/2737_mean                             +0.0054         +0.0054          2401          2414          2401          2414
advanced/execute/main/sha1_divs/5311_mean                             +0.0070         +0.0070          4403          4434          4403          4434
baseline/execute/main/sha1_divs/5311_mean                             +0.0093         +0.0093          4689          4733          4689          4733
advanced/execute/main/sha1_divs/65536_mean                            +0.0090         +0.0090         53577         54057         53577         54057
baseline/execute/main/sha1_divs/65536_mean                            +0.0050         +0.0050         57137         57423         57137         57424
advanced/execute/main/sha1_shifts/empty_mean                          +0.0004         +0.0004            32            32            32            32
baseline/execute/main/sha1_shifts/empty_mean                          +0.0032         +0.0032            35            35            35            35
advanced/execute/main/sha1_shifts/1351_mean                           +0.0012         +0.0013           668           669           668           669
baseline/execute/main/sha1_shifts/1351_mean                           +0.0040         +0.0040           741           744           741           744
advanced/execute/main/sha1_shifts/2737_mean                           -0.0011         -0.0011          1304          1302          1304          1302
baseline/execute/main/sha1_shifts/2737_mean                           +0.0035         +0.0035          1446          1451          1446          1451
advanced/execute/main/sha1_shifts/5311_mean                           -0.0023         -0.0023          2549          2543          2549          2543
baseline/execute/main/sha1_shifts/5311_mean                           +0.0034         +0.0034          2821          2830          2821          2830
advanced/execute/main/sha1_shifts/65536_mean                          -0.0064         -0.0064         31263         31062         31263         31062
baseline/execute/main/sha1_shifts/65536_mean                          -0.0003         -0.0003         34561         34552         34561         34552
advanced/execute/main/weierstrudel/0_mean                             +0.0113         +0.0113           169           171           169           171
baseline/execute/main/weierstrudel/0_mean                             +0.0029         +0.0029           171           172           171           172
advanced/execute/main/weierstrudel/1_mean                             +0.0082         +0.0082           374           378           374           378
baseline/execute/main/weierstrudel/1_mean                             +0.0096         +0.0096           382           385           382           385
advanced/execute/main/weierstrudel/3_mean                             +0.0092         +0.0092           585           591           585           591
baseline/execute/main/weierstrudel/3_mean                             +0.0036         +0.0036           599           602           599           602
advanced/execute/main/weierstrudel/9_mean                             +0.0140         +0.0140          1212          1229          1212          1229
baseline/execute/main/weierstrudel/9_mean                             +0.0025         +0.0025          1242          1245          1242          1245
advanced/execute/main/weierstrudel/14_mean                            +0.0121         +0.0121          1739          1760          1739          1760
baseline/execute/main/weierstrudel/14_mean                            +0.0013         +0.0013          1779          1781          1779          1782
advanced/execute/micro/beginsub_push1s_0xffff_mean                    -0.0143         -0.0143             0             0             0             0
baseline/execute/micro/beginsub_push1s_0xffff_mean                    +0.0022         +0.0022             0             0             0             0
advanced/execute/micro/beginsubs_0xffff_mean                          -0.0140         -0.0140             0             0             0             0
baseline/execute/micro/beginsubs_0xffff_mean                          +0.0026         +0.0026             0             0             0             0
advanced/execute/micro/jumpdests_0xffff_mean                          -0.0005         -0.0005           164           164           164           164
baseline/execute/micro/jumpdests_0xffff_mean                          +0.0006         +0.0006           108           108           108           108
advanced/execute/micro/loop_with_many_jumpdests_mean                  -0.0002         -0.0002         20019         20016         20019         20016
baseline/execute/micro/loop_with_many_jumpdests_mean                  -0.0003         -0.0003         13150         13147         13150         13147
advanced/execute/micro/push1s_0xffff_mean                             -0.0136         -0.0136             0             0             0             0
baseline/execute/micro/push1s_0xffff_mean                             +0.0028         +0.0028             0             0             0             0
advanced/execute/micro/push32s_0xffff_mean                            -0.0139         -0.0139             0             0             0             0
baseline/execute/micro/push32s_0xffff_mean                            +0.0022         +0.0022             0             0             0             0
advanced/execute/micro/zeros_0xffff_mean                              -0.0145         -0.0145             0             0             0             0
baseline/execute/micro/zeros_0xffff_mean
```

File: lib/evmone/execution_state.hpp [hunk 1]
```diff
@@ -101,11 +107,23 @@ struct ExecutionState
     evmc::HostContext host;
     evmc_revision rev = {};
     bytes return_data;
+
+    /// Reference to original EVM code.
+    /// TODO: Code should be accessed via code analysis only and this should be removed.
```
Comment (gumb0) [inline lib/evmone/execution_state.hpp:112]: Note: I use it in a somewhat hacky way in EOF1 implementation in Baseline to access the original (non-padded) code for `CODECOPY` / `CODESIZE`.
Don't know yet what would be a better approach (maybe keeping a reference to original code in analysis, too)

File: lib/evmone/baseline.cpp [hunk 2]
```diff
@@ -45,9 +45,10 @@ CodeAnalysis analyze(const uint8_t* code, size_t code_size)
 
 namespace
 {
-const uint8_t* op_jump(
-    ExecutionState& state, const CodeAnalysis::JumpdestMap& jumpdest_map) noexcept
+/// JUMP instruction implementation using baseline:CodeAnalysis.
```
Comment (gumb0) [inline lib/evmone/baseline.cpp:48]:
```suggestion
/// JUMP instruction implementation using baseline::CodeAnalysis.
```

**Review by gumb0 — APPROVED** [review]

File: lib/evmone/execution_state.hpp (hunk 1 above)
Comment (chfast) [reply lib/evmone/execution_state.hpp:112]: Yes, the original code will be available in the analysis.

**Pull request merged**